src/grpc/dp_grpc_conv.cpp

```diff
@@ -179,8 +192,23 @@ void DpToGrpcInterface(const struct dpgrpc_iface *dp_iface, Interface *grpc_ifac
 	grpc_iface->set_underlay_route(strbuf);
 }
 
+uint32_t DpIpv6MaskToPrefixLen(const uint8_t *mask) {
+	uint64_t high = ((uint64_t)mask[0] << 56) | ((uint64_t)mask[1] << 48) |
+					((uint64_t)mask[2] << 40) | ((uint64_t)mask[3] << 32) |
+					((uint64_t)mask[4] << 24) | ((uint64_t)mask[5] << 16) |
+					((uint64_t)mask[6] << 8)  | (uint64_t)mask[7];
+
+	uint64_t low = ((uint64_t)mask[8] << 56) | ((uint64_t)mask[9] << 48) |
+					((uint64_t)mask[10] << 40) | ((uint64_t)mask[11] << 32) |
+					((uint64_t)mask[12] << 24) | ((uint64_t)mask[13] << 16) |
+					((uint64_t)mask[14] << 8)  | (uint64_t)mask[15];
+
+	return __builtin_popcountll(high) + __builtin_popcountll(low);
+}
```

Comment:
Not sure if it would work without testing, but I would try `high = rte_be_to_cpu(*((uint64_t *)mask)` and equivalent `(mask+8)` for low.

Reply:
You mean smt like this:

>      uint64_t high = rte_be_to_cpu_64(*((const uint64_t *)(mask)));
>      uint64_t low = rte_be_to_cpu_64(*((const uint64_t *)(mask + 8)));

Seems to work in my single test case so I will convert it to this one as it seems more readable.